python/tests/api/writer/test_whylabs.py

```diff
@@ -395,3 +398,35 @@ def test_custom_client(self) -> None:
             writer.key_id
 
         assert len(WhylabsClientCache.instance()._api_client_cache) == 0
+
+    @patch(
+        "whylogs.api.writer.whylabs.WhyLabsWriter._get_upload_url_segmented_reference",
+        return_value=CreateReferenceProfileResponse(id="profile1", upload_urls=["http://upload.url"]),
+    )
+    @patch("whylogs.api.writer.whylabs.WhyLabsWriter._do_upload", return_value=(True, "Success"))
+    @patch("whylogs.api.writer.whylabs.WhyLabsWriter._get_dataset_timestamp", return_value=1234567890)
+    @patch("whylogs.api.writer.whylabs.WhyLabsWriter._upload_zipped_files", return_value=(True, "Success"))
+    def test_write_segmented_reference_result_set(
```

Comment:
I would probably leave this test out... it won't survive the refactoring.

Reply:
this "all mock" test has saved me some trouble, especially to understand exactly what are the method's calls. I'd vote for leaving it and just changing the patches -- or deleting it -- after the refactor